src/index.ts

```diff
@@ -43,15 +43,24 @@ class AbortError extends Error {
   }
 }
 
+interface ResourceLimits {
+  maxOldGenerationSizeMb? : number,
+  maxYoungGenerationSizeMb? : number,
+  codeRangeSizeMb? : number
+}
```

Comment:
```suggestion
type ResourceLimits = Worker extends {
  resourceLimits? : infer T;
} ? T : {};
```

(similar to what we’re doing to grab `TransferList` from the official TS definitions)